lib/carrierwave-aws.rb

```diff
@@ -43,7 +44,33 @@ def self.normalized_acl(acl)
     normalized
   end
 
+  def self.aws_signer(value = nil)
+    self.aws_signer = value if value
+
+    if instance_variable_defined?('@aws_signer')
+      @aws_signer
+    elsif superclass.respond_to? :aws_signer
+      superclass.aws_signer
+    end
+  end
+
+  def self.aws_signer=(signer)
+    @aws_signer = validated_signer(signer)
+  end
+
+  def self.validated_signer(signer)
+    unless signer.nil? || signer.instance_of?(Proc) && signer.arity == 2
```

Comment:
If somebody wants to detach a method from a module and try to pass that around they are on their own.